workflow/controller/controller.go

```diff
@@ -569,27 +594,33 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 	return true
 }
 
-func (wfc *WorkflowController) getPod(namespace string, podName string) (*apiv1.Pod, error) {
-	obj, exists, err := wfc.podInformer.GetStore().GetByKey(namespace + "/" + podName)
-	if err != nil {
-		return nil, err
-	}
-	if !exists {
-		return nil, nil
-	}
-	pod, ok := obj.(*apiv1.Pod)
-	if !ok {
-		return nil, fmt.Errorf("object is not a pod")
+func createFinalizerPodStatusRemovalPatchIfExists(pod *apiv1.Pod) []PatchOperation {
+	var patch []PatchOperation
+	i := slices.Index(pod.Finalizers, common.FinalizerPodStatus)
+	if i >= 0 {
+		patch = append(patch, PatchOperation{
+			Operation: "remove",
+			Path:      fmt.Sprintf("/metadata/finalizers/%d", i),
+		})
 	}
-	return pod, nil
+	return patch
 }
```

Comment:
Nit: Can we make this function more generic and pass in which finalizer it is? The side benefit is that we can make the name smaller, e.g.: `createFinalizerRemovalPatchIfExists`

Reply:
actually, why does this return a slice? why not just one?

Reply:
> actually, why does this return a slice? why not just one?

This implementation simply followed the original PR's implementation.
I have addressed this in the following commit.
https://github.com/argoproj/argo-workflows/pull/12413/commits/f96c17f3365bd5b90be08240cda436a70c177472